src/MongoDB/Driver/Cursor.cpp

```diff
@@ -323,6 +315,9 @@ void HHVM_METHOD(MongoDBDriverCursor, setTypeMap, const Array &typemap)
 	MongoDBDriverCursorData* data = Native::data<MongoDBDriverCursorData>(this_);
 
 	parseTypeMap(&data->bson_options, typemap);
+
+	/* Reconvert the current document to use the new typemap */
+	hippo_cursor_load_current(data);
```

Comment:
Is there a test or tracking ticket for this behavior? I'm pretty sure that PHPC does not recreate the current element when the type map is changed.